### PR TITLE
Handle disabled showVPNSettings

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/0/1205739952273970/f

### Description
See attached task description

### Steps to test this PR
https://app.asana.com/0/0/1205802524286063/f

### CODE REVIEW

**Comment by karlenDimla** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3650** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3650" target="_blank">Graphite</a> 
    * **PR #3752** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3752" target="_blank">Graphite</a>  👈
    * **PR #3673** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3673" target="_blank">Graphite</a> 
      * **PR #3714** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3714" target="_blank">Graphite</a> 
        * **PR #3719** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3719" target="_blank">Graphite</a> 
      * **PR #3674** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3674" target="_blank">Graphite</a> 
        * **PR #3723** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3723" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3752?utm_source=stack-comment).

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt [hunk 1]
```diff
@@ -68,6 +73,8 @@ class RealWgServerApi @Inject constructor(
         val userPreferredLocation = netPGeoswitchingRepository.getUserPreferredLocation()
         val registerKeyBody = if (selectedServer != null) {
             RegisterKeyBody(publicKey = publicKey, server = selectedServer)
+        } else if (!netPRemoteFeature.showVpnSettings().isEnabled() && !appBuildConfig.isInternalBuild()) {
+            RegisterKeyBody(publicKey = publicKey, server = "*")
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt:77]: Do we need this check. I guess I know what you're trying to do, but I don't think it fully achieves the goal due to the previous IF, and if settings are disabled users won't get access to geoswitching anyways

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/NetPNotificationSettingsActivity.kt [hunk 2]
```diff
@@ -72,5 +72,5 @@ class NetPNotificationSettingsActivity : DuckDuckGoActivity() {
 }
 
 internal object NetPNotificationSettingsScreenNoParams : GlobalActivityStarter.ActivityParams {
-    private fun readResolve(): Any = NetPVpnSettingsScreenNoParams
+    private fun readResolve(): Any = NetPNotificationSettingsScreenNoParams
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/NetPNotificationSettingsActivity.kt:75]: nice catch

Comment (karlenDimla) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/NetPNotificationSettingsActivity.kt:75]: I’ll fix this somewhere else

**Pull request closed without merging**